src/main/java/seedu/guilttrip/logic/commands/remindercommands/SetReminderCommand.java

```diff
@@ -0,0 +1,97 @@
+package seedu.guilttrip.logic.commands.remindercommands;
+
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_DESC;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_FREQ;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_PERIOD;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import java.util.List;
+
+import seedu.guilttrip.commons.core.index.Index;
+import seedu.guilttrip.commons.util.TimeUtil;
+import seedu.guilttrip.logic.CommandHistory;
+import seedu.guilttrip.logic.commands.Command;
+import seedu.guilttrip.logic.commands.CommandResult;
+import seedu.guilttrip.logic.commands.exceptions.CommandException;
+import seedu.guilttrip.model.Model;
+import seedu.guilttrip.model.entry.Date;
+import seedu.guilttrip.model.entry.Description;
+import seedu.guilttrip.model.entry.Expense;
+import seedu.guilttrip.model.entry.Income;
+import seedu.guilttrip.model.entry.Period;
+import seedu.guilttrip.model.entry.Wish;
+import seedu.guilttrip.model.reminders.IewReminder;
+import seedu.guilttrip.model.reminders.Reminder;
+import seedu.guilttrip.model.util.Frequency;
+
+/**
+ * Set a reminder for an expense/ income/ wish.
+ */
+public class SetReminderCommand extends Command {
+    public static final String COMMAND_WORD = "setReminder";
+    public static final String UNSUPPORTED_TYPE = "Unable to create reminder for this entry.";
+    public static final String INVALID_PERIOD = "Invalid period. (Cannot set reminder to notify before today!)";
+    public static final String INVALID_FREQ = "Frequency must be smaller than period.";
+    public static final String MESSAGE_SUCCESS = "New reminder added for: %1$s";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
```

Comment:
Can you refactor the one-liner description into a variable? And add it into the map function in `ParserUtil.java` if you want your command to be reflected in the AutoSuggester.

You can check out any other command for an example!

Reply:
Actually if you check my Commands already have command word one liner and message usage.